src/LayoutManager/LayoutManager.ts

```diff
@@ -0,0 +1,184 @@
+import { Point } from '../Point';
+import { CENTER } from '../constants';
+import type { FabricObject } from '../shapes/Object/FabricObject';
+import { invertTransform } from '../util/misc/matrix';
+import { FitContentLayout } from './LayoutStrategies/FitContentLayout';
+import type { LayoutStrategy } from './LayoutStrategies/LayoutStrategy';
+import type { LayoutContext, LayoutResult, StrictLayoutContext } from './types';
+
+export class LayoutManager {
+  private _firstLayoutDone = false;
+  private _prevLayoutStrategy?: LayoutStrategy;
+
+  strategy: LayoutStrategy;
+
+  constructor(strategy: LayoutStrategy = new FitContentLayout()) {
+    this.strategy = strategy;
+  }
+
+  performLayout(context: LayoutContext) {
+    if (!this._firstLayoutDone && context.type !== 'initialization') {
+      //  reject layout requests before initialization layout
+      return;
+    }
+
+    const strictContext: StrictLayoutContext = {
+      strategy: this.strategy,
+      prevStrategy: this._prevLayoutStrategy,
+      strategyChange:
+        !!this._prevLayoutStrategy &&
+        this.strategy !== this._prevLayoutStrategy,
+      ...context,
+    };
+
+    this.onBeforeLayout(strictContext);
+
+    const layoutResult = this.getLayoutResult(strictContext);
+    let bubblingContext: LayoutResult | undefined;
+
+    if (!this._firstLayoutDone) {
+      if (layoutResult) {
+        this.commitLayout(strictContext, layoutResult);
```

Comment:
It seems like:

```ts
this.commitLayout(strictContext, layoutResult);
bubblingContext = layoutResult;
```

is called regardless of `this._firstLayoutDone` as far as `layoutResult` is defined. So it could be extracted before?

Reply:
will do